src/global_mgr.cxx

```diff
@@ -24,10 +24,41 @@ limitations under the License.
 #include "raft_server.hxx"
 #include "tracer.hxx"
 
+#include <memory>
+
 namespace nuraft {
 
-std::atomic<nuraft_global_mgr*> nuraft_global_mgr::instance_(nullptr);
-std::mutex nuraft_global_mgr::instance_lock_;
+class ngm_singleton {
+public:
+    static ngm_singleton& get_instance() {
+        static ngm_singleton instance;
+        return instance;
+    }
+
+    nuraft_global_mgr* get() {
+        return internal_.get();
+    }
+
+    bool create() {
+        if (internal_.get()) {
+            // Already created.
+            return false;
+        }
+        // C++11 doesn't have `make_unique`.
```

Comment:
Maybe just implement it?)

Reply:
Using constructor with raw pointer is ok for now :)